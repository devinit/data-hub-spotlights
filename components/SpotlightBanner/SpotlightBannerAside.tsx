import React, { FunctionComponent } from 'react';

interface SpotlightBannerAsideProps {
  className: string;
}

const SpotlightBannerAside: FunctionComponent<SpotlightBannerAsideProps> = ({ children, className }) => (
  <div className={className}>{children}</div>
);

export { SpotlightBannerAside };
