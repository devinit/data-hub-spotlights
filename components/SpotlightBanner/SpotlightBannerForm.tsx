import React, { FunctionComponent } from 'react';

const SpotlightBannerForm: FunctionComponent = ({ children }) => {
  return (
    <div className="spotlight-banner__form">{ children }</div>
  );
};

export { SpotlightBannerForm };
