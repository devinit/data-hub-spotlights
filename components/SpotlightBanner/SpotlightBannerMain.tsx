import React, { FunctionComponent } from 'react';

const SpotlightBannerMain: FunctionComponent = ({ children }) =>
  <div className="spotlight-banner__main">{ children }</div>;

export { SpotlightBannerMain };
