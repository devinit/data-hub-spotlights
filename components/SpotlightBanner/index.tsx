export * from './SpotlightBanner';
export * from './SpotlightBannerAside';
export * from './SpotlightBannerMain';
export * from './SpotlightBannerForm';
