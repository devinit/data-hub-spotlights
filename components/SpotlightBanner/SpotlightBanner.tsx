import classNames from 'classnames';
import React, { Children, FunctionComponent, isValidElement, ReactNode } from 'react';
import { SpotlightBannerAside } from './SpotlightBannerAside';
import { SpotlightBannerMain } from './SpotlightBannerMain';

const SpotlightBanner: FunctionComponent<{ header?: boolean }> = ({ children, header }) => {
  const renderValidChildren = (): ReactNode => {
    return Children.map(children, child => {
      if (isValidElement(child) && (child.type === SpotlightBannerAside || child.type === SpotlightBannerMain)) {
        return child;
      }
    });
  };

  return (
    <div className={classNames('spotlight-banner', { 'spotlight-banner--header': header })}>
      {renderValidChildren()}
    </div>
  );
};

SpotlightBanner.defaultProps = {
  header: false
};

export { SpotlightBanner };
