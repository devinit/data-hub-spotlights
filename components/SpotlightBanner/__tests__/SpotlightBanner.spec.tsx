/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { SpotlightBanner } from '../SpotlightBanner';
import { SpotlightBannerAside } from '../SpotlightBannerAside';
import { SpotlightBannerMain } from '../SpotlightBannerMain';

describe('SpotlightBanner', () => {
  test('renders correctly', () => {
    const renderer = TestRenderer.create(<SpotlightBanner />).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('renders children correctly', () => {
    const renderer = TestRenderer.create(
      <SpotlightBanner>
        <SpotlightBannerAside />
        <SpotlightBannerMain />
        <div>My Other Child</div>
      </SpotlightBanner>
    ).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('renders the class spotlight-banner--header when configured as a header', () => {
    const renderer = TestRenderer.create(<SpotlightBanner header />).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
