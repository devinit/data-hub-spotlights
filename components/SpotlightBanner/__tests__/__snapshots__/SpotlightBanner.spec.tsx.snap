// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpotlightBanner renders correctly 1`] = `
<div
  className="spotlight-banner"
/>
`;

exports[`SpotlightBanner renders only SpotlightBannerAside & SpotlightBannerMain 1`] = `
<div
  className="spotlight-banner"
>
  <div
    className="spotlight-banner__aside"
  />
  <div
    className="spotlight-banner__main"
  />
</div>
`;

exports[`SpotlightBanner renders the class spotlight-banner--header when configured as a header 1`] = `
<div
  className="spotlight-banner spotlight-banner--header"
/>
`;
